internal/install/application_configuration.go

```diff
@@ -167,15 +168,36 @@ func selectElasticAgentImageName(version, agentBaseImage string) string {
 		disableWolfiImages = true
 	}
 	switch {
-	case !disableWolfiImages && !v.LessThan(elasticAgentWolfiVersion) && agentBaseImage != "complete":
-		return elasticAgentWolfiImageName
-	case !v.LessThan(elasticAgentCompleteOwnNamespaceVersion):
+	case agentBaseImage == "complete":
+		return selectElasticAgentCompleteImageName(v)
+	case agentBaseImage == "systemd":
+		return selectElasticAgentSystemDImageName(v)
+	default:
+		switch {
+		case !disableWolfiImages && !v.LessThan(elasticAgentWolfiVersion):
+			return elasticAgentWolfiImageName
+		default:
+			return selectElasticAgentCompleteImageName(v)
+		}
```

Comment:
Keep the same behaviour as it was previously. It means using the complete image whenever it is possible.